pkg/capture/outputlocation/s3.go

```diff
@@ -0,0 +1,184 @@
+		su.l.Debug("Output location is not enabled because both endpoint and region are not set", zap.String("location", su.Name()))
+		return false
+	}
+
+	su.path = os.Getenv(string(captureConstants.CaptureOutputLocationEnvKeyS3Path))
+
+	var err error
+	su.accessKeyID, err = readAccessKeyID()
+	if err != nil {
+		su.l.Error("Failed to obtain access key id from secret", zap.Error(err))
+		return false
+	}
+
+	su.secretAccessKey, err = readSecretAccessKey()
+	if err != nil {
+		su.l.Error("Failed to obtain secret access key id from secret", zap.Error(err))
+		return false
+	}
+
+	return true
+}
+
+func (su *S3Upload) Output(srcFilePath string) error {
+	objectKey := path.Join(su.path, srcFilePath)
+
+	su.l.Info("Upload capture file to s3",
+		zap.String("location", su.Name()),
+		zap.String("source file path", srcFilePath),
+		zap.String("bucketName", su.bucket),
+		zap.String("objectKey", objectKey),
+	)
+
+	s3Client, err := su.getClient()
+	if err != nil {
+		su.l.Error("Failed to get AWS client", zap.Error(err))
+		return err
+	}
+
+	s3File, err := os.Open(srcFilePath)
+	if err != nil {
+		wrappedErr := fmt.Errorf("failed to open src file %s: %w", srcFilePath, err)
+		su.l.Error("Failed to open capture file", zap.Error(wrappedErr))
+		return wrappedErr
+	}
+	defer s3File.Close()
+
+	_, err = s3Client.PutObject(context.TODO(), &s3.PutObjectInput{
+		Bucket: aws.String(su.bucket),
+		Key:    aws.String(objectKey),
+		Body:   s3File,
+	})
+	if err != nil {
+		wrappedErr := fmt.Errorf("failed to upload file to S3: %w", err)
+		su.l.Error("Couldn't upload file",
+			zap.String("srcFilePath", srcFilePath),
+			zap.String("bucketName", su.bucket),
+			zap.String("objectKey", objectKey),
+			zap.Error(wrappedErr))
+	}
+	return nil
+}
+
+func (su *S3Upload) getClient() (*s3.Client, error) {
+	var opts []func(options *config.LoadOptions) error
+
+	if su.endpoint != "" {
+		opts = append(opts,
+			config.WithEndpointResolverWithOptions(
+				aws.EndpointResolverWithOptionsFunc(func(service, region string, options ...interface{}) (aws.Endpoint, error) {
+					return aws.Endpoint{
+						URL:               su.endpoint,
+						HostnameImmutable: true,
+					}, nil
+				}),
+			),
+		)
+	}
+
+	if su.region != "" {
+		opts = append(opts, config.WithRegion(su.region))
+	} else {
+		opts = append(opts, config.WithRegion("auto"))
+	}
+
+	if su.accessKeyID != "" {
+		opts = append(opts, config.WithCredentialsProvider(credentials.NewStaticCredentialsProvider(
+			su.accessKeyID,
+			su.secretAccessKey,
+			"",
+		)))
+	}
+
+	cfg, err := config.LoadDefaultConfig(context.TODO(),
+		opts...,
+	)
+	if err != nil {
+		return nil, fmt.Errorf("failed to load AWS config: %w", err)
+	}
+
+	return s3.NewFromConfig(cfg), nil
+}
+
+func readAccessKeyID() (string, error) {
+	secretPath := filepath.Join(captureConstants.CaptureOutputLocationS3UploadSecretPath, captureConstants.CaptureOutputLocationS3UploadAccessKeyID)
+	if runtime.GOOS == "windows" {
```

Comment:
I'm curious about why there's platform-specific behavior here.

Reply:
I implemented the code by referring to outputlocation/blob.go. I haven’t tested this on Windows. But I think we might need to set the secret path using the `CONTAINER_SANDBOX_MOUNT_POINT` env, considering that the absolute paths for volume mounts can differ in Windows.

https://kubernetes.io/docs/tasks/configure-pod-container/create-hostprocess-pod/#volume-mounts

https://github.com/microsoft/retina/blob/44a2a59c9d2f6dc02d578c503dc44cffe5eba2f9/pkg/capture/outputlocation/blob.go#L89-L100